app/audit/models/OptOutAuditModel.scala

```diff
@@ -0,0 +1,113 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package audit.models
+
+import auth.MtdItUser
+import connectors.itsastatus.ITSAStatusUpdateConnectorModel.{ITSAStatusUpdateResponse, ITSAStatusUpdateResponseFailure, ITSAStatusUpdateResponseSuccess}
+import models.incomeSourceDetails.TaxYear
+import models.itsaStatus.ITSAStatus.ITSAStatus
+import play.api.libs.json.{JsObject, JsValue, Json, OFormat}
+import services.optout.OptOutProposition
+
+case class Outcome(
+                    isSuccessful: Boolean,
+                    failureCategory: Option[String],
+                    failureReason: Option[String]
+                  )
+
+
+object Outcome {
+  implicit val format: OFormat[Outcome] = Json.format[Outcome]
+}
+
+case class OptOutAuditModel(
+                             //                             mtdItUser: MtdItUser[_],
+                             nino: String,
+                             outcome: Outcome,
+                             optOutRequestedFromTaxYear: String,
+                             currentYear: String,
+                             beforeITSAStatusCurrentYearMinusOne: ITSAStatus,
+                             beforeITSAStatusCurrentYear: ITSAStatus,
+                             beforeITSAStatusCurrentYearPlusOne: ITSAStatus,
+                             afterAssumedITSAStatusCurrentYearMinusOne: ITSAStatus,
+                             afterAssumedITSAStatusCurrentYear: ITSAStatus,
+                             afterAssumedITSAStatusCurrentYearPlusOne: ITSAStatus,
+                             currentYearMinusOneCrystallised: Boolean
+                           ) extends ExtendedAuditModel {
+
+
+  override val transactionName: String = enums.TransactionName.OptOutQuarterlyReportingRequest
+
+  override val auditType: String = enums.AuditType.OptOutQuarterlyReportingRequest
+
+  private val optOutDetailsJson: JsObject =
+    Json.obj(
+      "nino" -> nino,
+      "outcome" -> outcome,
+      "optOutRequestedFromTaxYear" -> optOutRequestedFromTaxYear,
+      "currentYear" -> currentYear,
+      "beforeITSAStatusCurrentYearMinusOne" -> beforeITSAStatusCurrentYearMinusOne.toString,
+      "beforeITSAStatusCurrentYear" -> beforeITSAStatusCurrentYear.toString,
+      "beforeITSAStatusCurrentYearPlusOne" -> beforeITSAStatusCurrentYearPlusOne.toString,
+      "afterAssumedITSAStatusCurrentYearMinusOne" -> afterAssumedITSAStatusCurrentYearMinusOne.toString,
+      "afterAssumedITSAStatusCurrentYear" -> afterAssumedITSAStatusCurrentYear.toString,
+      "afterAssumedITSAStatusCurrentYearPlusOne" -> afterAssumedITSAStatusCurrentYearMinusOne.toString,
+      "currentYearMinusOneCrystallised" -> currentYearMinusOneCrystallised
```

Comment:
I noticed that in the TXM proposed audit event document it shows the field names including `-1` and `+1` instead of `MinusOne` and `PlusOne` shown in the code.

Preferably it should stay as words but i wasn't sure if it had to be exact or not.

Reply:
Yeah i think we need some clarification on this because personally i don't really like having `-1` and `+1` as part of the field names but that's currently what the proposed json body says

Reply:
just to note what was mentioned in slack we've decided to implement the field names as the ones in confluence.